pentasharp/Controllers/BusSchedule.cs

```diff
@@ -1,21 +1,336 @@
+        {
+            var route = _context.BusRoutes.Find(id);
+            if (route == null)
+                return NotFound(new { success = false, message = "Route not found." });
+
+            _context.BusRoutes.Remove(route);
+            _context.SaveChanges();
+
+            return Ok(new { success = true, message = "Route deleted successfully." });
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpPost("AddSchedule")]
+        public IActionResult AddSchedule([FromBody] AddScheduleViewModel model)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(new { success = false, message = "Invalid data provided." });
+
+            var route = _context.BusRoutes.FirstOrDefault(r => r.RouteId == model.RouteId);
+            if (route == null)
+                return BadRequest(new { success = false, message = "Route not found." });
+
+            var bus = _context.Buses.FirstOrDefault(b => b.BusId == model.BusId);
+            if (bus == null)
+                return BadRequest(new { success = false, message = "Bus not found." });
+
+            if (_context.BusSchedules.Any(s => s.BusId == model.BusId && s.DepartureTime == model.DepartureTime))
+                return BadRequest(new { success = false, message = "A schedule for the same bus at the same time already exists." });
+
+            model.ArrivalTime = model.DepartureTime.Add(route.EstimatedDuration);
+            model.AvailableSeats = bus.Capacity;
+
+            var schedule = _mapper.Map<BusSchedule>(model);
+            schedule.Status = BusScheduleStatus.Scheduled;
+
+            _context.BusSchedules.Add(schedule);
+            _context.SaveChanges();
+
+            return Ok(new { success = true, message = "Schedule added successfully." });
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpGet("GetSchedules")]
+        public IActionResult GetSchedules()
+        {
+            var schedules = _context.BusSchedules.Include(s => s.Route).Include(s => s.Bus).ToList();
+            var scheduleViewModels = schedules.Select(s => new
+            {
+                s.ScheduleId,
+                s.RouteId,
+                s.BusId,
+                s.DepartureTime,
+                s.ArrivalTime,
+                s.Price,
+                s.AvailableSeats,
+                Status = s.Status.ToString()
+            }).ToList();
+
+            return Ok(scheduleViewModels);
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpPut("EditSchedule/{id}")]
+        public IActionResult EditSchedule(int id, [FromBody] AddScheduleViewModel model)
+        {
+            var schedule = _context.BusSchedules.Find(id);
+            if (schedule == null)
+                return NotFound(new { success = false, message = "Schedule not found." });
+
+            if (_context.BusSchedules.Any(s => s.ScheduleId != id && s.BusId == model.BusId && s.DepartureTime == model.DepartureTime))
```

Comment:
separate into a local variable and then call the variable here, remember to use linq